staging/src/k8s.io/apiserver/pkg/apis/apiserver/types.go

```diff
@@ -166,3 +166,46 @@ type TracingConfiguration struct {
 	// Defaults to 0.
 	SamplingRatePerMillion *int32
 }
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// PriorityAndFairnessConfiguration holds API Priority and Fairness parameters.
+type PriorityAndFairnessConfiguration struct {
+	metav1.TypeMeta
+
+	// WorkEstimatorConfig holds work estimator parameters.
+	WorkEstimator WorkEstimatorConfiguration
+}
+
+// WorkEstimatorConfiguration holds work estimator parameters.
+type WorkEstimatorConfiguration struct {
+	// ListWorkEstimator holds work estimator parameters related to list requests.
+	ListWorkEstimator ListWorkEstimatorConfiguration
+	// MutatingWorkEstimator holds work estimator parameters
+	// related to requests that mutate watched objects.
+	MutatingWorkEstimator MutatingWorkEstimatorConfiguration
+
+	// MaximumSeats is the maximum number of seats a request can occupy
+	MaximumSeats uint64
+}
+
+// ListWorkEstimatorConfiguration holds work estimator parameters related to list requests.
+type ListWorkEstimatorConfiguration struct {
+	// ObjectsPerSeat is a number of objects from a single list request per seat.
+	ObjectsPerSeat float64
+}
+
+// MutatingWorkEstimatorConfiguration holds work estimator
+// parameters related to watches of mutating objects.
+type MutatingWorkEstimatorConfiguration struct {
+	// EventAdditionalDuration is the additional amount of time that a mutating request occupies seats
+	// to account for the work involved in sending the corresponding WATCH notifications. Setting its
+	// value to 0 effectively disables this behavior but a request may still occupy seats.
+	EventAdditionalDuration metav1.Duration
```

Comment:
I may have come across a use case for something like this -- when scheduler writes to /binding, it implies ~7 additional writes from other processes across the cluster will come in shortly. It'd be handy to artificially increase the duration of such requests so that the scheduler doesn't produce a backlog of requests that apiserver can't service.

This is a good argument for not specifically mentioning "events" in the name of this field, at least.

cc @alculquicondor

Reply:
+1